pkgs/development/python-modules/whisperx/default.nix

```diff
@@ -0,0 +1,87 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+
+  setuptools,
+
+  ctranslate2,
+  ctranslate2-cpp,
+  faster-whisper,
+  ffmpeg,
+  nltk,
+  pandas,
+  pyannote-audio,
+  torch,
+  torchaudio,
+  transformers,
+
+  cudaSupport ? torch.cudaSupport,
+  cudaPackages,
+}:
+
+let
+  # inherit (cudaPackages) cudaFlags cudnn nccl;
+
+  ctranslate = ctranslate2.override {
+    ctranslate2-cpp = ctranslate2-cpp.override {
+      withCUDA = cudaSupport ? false;
+      withCuDNN = cudaSupport ? false;
+    };
+  };
+in
+buildPythonPackage rec {
+  pname = "whisperx";
+  version = "3.3.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "m-bain";
+    repo = "whisperX";
+    tag = "v${version}";
+    hash = "sha256-Wb9jKTs0rTdgjvcnKl1P8Uh6b5jzlkKgQBcJYA+7+W4=";
+  };
+
+  nativeBuildInputs = lib.optionals cudaSupport ( with cudaPackages; [
+
+  ]);
```

Comment:
?

Reply:
Sorry, these were WIP commits, I didnt mean to push this.